rpc/dialer.go

```diff
@@ -79,6 +83,17 @@ type GrpcOverHTTPClientConn struct {
 	*grpc.ClientConn
 }
 
+// GRPCClientConnInterface is the allow list of methods
+// we expect to be able to call on a *grpc.ClientConn.
+type GRPCClientConnInterface interface {
```

Comment:
I decided to return an interface so that we can control what subset of the grpc.ClientConn methods we want to sign up to support for callers of the goutils rpc package.

It also allows us to mock out the connectivity.State in tests which is useful.